File: java/gate 6/MiniParkingSystem.java
import java.util.Scanner;

public class MiniParkingSystem {

     int totalSpaces;
    int occupiedSpaces;

    
    public void initialize(int space) {
        totalSpaces = space;
        occupiedSpaces = 0;
    }

   
    public boolean parkCar(int carNumber) {
        if (occupiedSpaces < totalSpaces) {
            occupiedSpaces++;
            return true;
        }
        return false;
    }

    
    public boolean removeCar(int carNumber) {
        if (occupiedSpaces > 0) {
            occupiedSpaces--;
            return true;
        }
        return false;
    }

    
    public int checkAvailability() {
        return totalSpaces - occupiedSpaces;
    }

    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter your name: ");
        String name = scanner.nextLine();

        System.out.print("Enter your car number: ");
        int carNumber = scanner.nextInt();

        System.out.print("Enter your preferred parking space between 1 and 20: ");
        int parkingSpace = scanner.nextInt();

        int totalSpaces = 20;

        if (parkingSpace < 1 || parkingSpace > totalSpaces) {
            System.out.println("Invalid choice");
            return;
        }

        System.out.println("\n" + name + "!");

        MiniParkingSystem lot = new MiniParkingSystem();
        lot.initialize(totalSpaces);

        boolean result = lot.parkCar(carNumber);

        if (!result) {
            System.out.println("Parking failed");
        } else {
            System.out.println("Car parked successfully in space " + parkingSpace);
        }
    }
}

File: python/gate 6/mini_parking_system.py
def parking_spaces(space):
    return space[0]


def car_is_parked_successfully(space, cars):
    if cars <= space[0]:
        return cars
    return -1


def cannot_park_in_an_occupied_spot(space, cars):
    if cars > space[0]:
        return -1
    return cars


def car_is_removed(space, cars):
    return space[0]


def cannot_remove_car_from_an_empty_spot(space, cars):
    return -1


def cannot_remove_a_car_from_a_lot_you_did_not_park(space, cars):
    return -1


def check_parking_spaces_availability(space, parked, leaving, new_cars):
    return space[0] - parked + leaving - new_cars


def run_parking_system():
    name = input("Enter your name: ")
    car_number = int(input("Enter your car number: "))
    parking_space = int(input("Enter your preferred parking space between 1 and 20: "))

    total_spaces = 20

    
    if parking_space < 1 or parking_space > total_spaces:
        print("Invalid choice")
        return

    print(f"\n {name}!")
   

    
    result = car_is_parked_successfully([total_spaces], 1)

    if result == -1:
        print("Parking failed")
    else:
        print(f"Car parked successfully in space {parking_space}")


